src/main/java/com/gravity9/mongocdc/MongoChangeStreamWorker.java

```diff
@@ -162,10 +162,19 @@ public void run() {
         } while (true);
     }
 
-    void register(ChangeStreamListener listener) {
-        log.info("Registering listener {} to worker on partition {} for collection {}", listener.getClass().getName(), partition, mongoConfig.getCollectionName());
-        listeners.add(listener);
-    }
+	void register(ChangeStreamListener listener) {
+		log.info("Registering listener {} to worker on partition {} for collection {}", listener.getClass().getName(), partition, mongoConfig.getCollectionName());
+		listeners.add(listener);
+	}
+
+	public void deregister(ChangeStreamListener listener) {
+		log.info("Unregistering listener {} from worker on partition {} for collection {}", listener.getClass().getName(), partition, mongoConfig.getCollectionName());
+		listeners.remove(listener);
```

Comment:
To support this line `listeners` must be a synchronized collection.

Reply:
done. Used `CopyOnWriteArraySet` as you proposed